src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,71 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { resetCurrentTodo, selectCurrentTodo } from '../../features';
 
 export const TodoModal: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [todoUser, setTodoUser] = useState<User | null>(null);
+  const dispatch = useAppDispatch();
+
+  const activeTodo = useAppSelector(selectCurrentTodo);
+
+  useEffect(() => {
+    if (activeTodo) {
+      getUser(activeTodo.userId)
+        .then(setTodoUser)
+        .finally(() => setIsLoading(false));
+    }
```

Comment:
The `setIsLoading(false)` is called in the `finally` block, which means it will execute even if `getUser` fails. Consider handling errors explicitly to ensure that the loading state reflects the actual data fetching status. You might want to set `isLoading` to false only after successfully fetching the user data or handle errors appropriately.